code/game/machinery/cryopod.dm

```diff
@@ -466,56 +450,65 @@ var/global/list/frozen_items = list("Alpha"=list(),"Bravo"=list(),"Charlie"=list
 	set category = "Object"
 	set src in oview(1)
 
-	if(usr.stat != 0 || !(ishuman(usr) || ismonkey(usr)))
+	if(usr.stat != CONSCIOUS || !(ishuman(usr) || ismonkey(usr)))
 		return
 
 	if(occupant)
 		to_chat(usr, "<span class='warning'>[src] is occupied.</span>")
 		return
 
-	if(isxeno(usr))
-		to_chat(usr, "<span class='warning'>There is no way [src] will accept you!</span>")
-		return
+	climb_in(usr)
 
+/obj/machinery/cryopod/proc/climb_in(mob/user)
 	usr.visible_message("<span class='notice'>[usr] starts climbing into [src].</span>",
 	"<span class='notice'>You start climbing into [src].</span>")
 
-	if(do_after(usr, 20, FALSE, 5, BUSY_ICON_GENERIC))
-
-		if(!usr || !usr.client)
-			return
-
-		if(occupant)
-			to_chat(usr, "<span class='warning'>[src] is occupied.</span>")
-			return
-
-		usr.forceMove(src)
-		occupant = usr
+	if(!do_after(usr, 20, FALSE, 5, BUSY_ICON_GENERIC))
+		return
 
-		if(orient_right)
-			icon_state = "body_scanner_1-r"
-		else
-			icon_state = "body_scanner_1"
+	if(!usr?.client)
+		return
 
-		to_chat(usr, "<span class='notice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
-		to_chat(usr, "<span class='boldnotice'>If you ghost, log out or close your client now, your character will shortly be permanently removed from the round.</span>")
-		time_entered = world.time
-		start_processing()
+	if(occupant)
+		to_chat(usr, "<span class='warning'>[src] is occupied.</span>")
+		return
 
-		add_fingerprint(usr)
+	usr.forceMove(src)
+	occupant = usr
+	update_icon()
 
-	return
+	to_chat(usr, "<span class='notice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
+	to_chat(usr, "<span class='boldnotice'>If you ghost, log out or close your client now, your character will shortly be permanently removed from the round.</span>")
+	time_entered = world.time
+	start_processing()
+	add_fingerprint(usr)
 
 /obj/machinery/cryopod/proc/go_out()
 
 	if(!occupant)
 		return
 
-	occupant.forceMove(get_turf(src))
+	//Eject any items that aren't meant to be in the pod.
+	var/list/items = contents - announce
+
+	for(var/atom/movable/A in items)
+		occupant.forceMove(get_turf(src))
+
 	occupant = null
 	stop_processing()
-
-	if(orient_right)
-		icon_state = "body_scanner_0-r"
-	else
-		icon_state = "body_scanner_0"
+	update_icon()
+
+/obj/machinery/cryopod/admin //Invisible admin magic.
```

Comment:
This is 100x better, but the next step up would me to make it a mob proc so there's no need to spawn a machine.